Organization Hooks aren't Typed

### Is this suited for github?

- [X] Yes, this is suited for github

### To Reproduce

Add the organization plugin to your config 

```typescript
  plugins: [openAPI(), passkey(), nextCookies(), organization(), adminPlugin()],
```

add organizationClient to your client 

```typescript
export const authClient = createAuthClient({
  plugins: [
    passkeyClient(),
    magicLinkClient(),
    organizationClient(),
    adminClient(),
  ],
});
```

add to your auth 

```typescript
import { betterAuth } from "better-auth";
import { authConfig } from "@/lib/auth/config";
import { authActions } from "@/lib/auth/actions";

export const auth = betterAuth({
  ...authConfig,
  ...authActions,
});
```

### Current vs. Expected behavior

Currently the following hooks are Any typed. I would expect them to have the appropriate types for the organization plugin. 

![image](https://github.com/user-attachments/assets/e37f94a3-752d-4116-9eee-93c486cca571)


### What version of Better Auth are you using?

1.1.9

### Provide environment information

```bash
- Windows 11
```


### Which area(s) are affected? (Select all that apply)

Types

### Auth config (if applicable)

```typescript
import { drizzleAdapter } from "better-auth/adapters/drizzle";
import { db } from "@/db";
import {
  openAPI,
  organization,
  admin as adminPlugin,
} from "better-auth/plugins";
import { passkey } from "better-auth/plugins/passkey";
import { APP_NAME } from "@/utils/constants";
import { env } from "@/env";
import { BetterAuthOptions } from "better-auth";
import { nextCookies } from "better-auth/next-js";

export const authConfig = {
  appName: APP_NAME,
  baseURL:
    env.NODE_ENV === "production"
      ? env.NEXT_PUBLIC_APP_URL
      : "http://app.localhost:3000",
  trustedOrigins: [
    env.NEXT_PUBLIC_APP_URL,
    "http://app.localhost:3000",
    "http://localhost:3000",
  ],
  logger: {
    disabled: env.NODE_ENV === "production",
    level: "debug",
  },
  database: drizzleAdapter(db, {
    provider: "pg",
  }),
  session: {
    freshAge: 0,
    expiresIn: 60 * 60 * 24 * 3, // 3 days
    updateAge: 60 * 60 * 12, // 12 hours (every 12 hours the session expiration is updated)
    cookieCache: {
      enabled: true,
      maxAge: 60 * 5, // 5 minutes
    },
  },
  user: {
    additionalFields: {
      defaultOrganization: {
        type: "string",
        required: false,
        input: false,
      },
    },
    changeEmail: {
      enabled: true,
    },
    deleteUser: {
      enabled: true,
    },
  },
  emailAndPassword: {
    enabled: true,
    requireEmailVerification: true,
  },
  socialProviders: {
    microsoft: {
      clientId: env.MICROSOFT_CLIENT_ID,
      clientSecret: env.MICROSOFT_CLIENT_SECRET,
      getUserInfo: async (token) => {
        const res = await fetch("https://graph.microsoft.com/v1.0/me", {
          headers: {
            Authorization: `Bearer ${token.accessToken}`,
          },
        });
        const user = await res.json();
        console.log(user);

        return {
          user: {
            id: user.id,
            name: user.displayName,
            email: user.userPrincipalName,
            emailVerified: true,
            createdAt: new Date(),
            updatedAt: new Date(),
          },
          data: user.data,
        };
      },
    },
  },

  plugins: [openAPI(), passkey(), nextCookies(), organization(), adminPlugin()],
} satisfies BetterAuthOptions;
```


### Additional context

_No response_

Still fails in ci though. 

```bash
[lint] next lint exited with code 0
[format] All matched files use Prettier code style!
[format] pnpm run format:check exited with code 0
Error: [typecheck] components/layout/sidebar/cmdk-org-switcher.tsx(144,41): error TS7006: Parameter 'org' implicitly has an 'any' type.
[typecheck] tsc --noEmit exited with code 2
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

> Still fails in ci though.
> 
> ```shell
> [lint] next lint exited with code 0
> [format] All matched files use Prettier code style!
> [format] pnpm run format:check exited with code 0
> Error: [typecheck] components/layout/sidebar/cmdk-org-switcher.tsx(144,41): error TS7006: Parameter 'org' implicitly has an 'any' type.
> [typecheck] tsc --noEmit exited with code 2
>  ELIFECYCLE  Command failed with exit code 1.
> Error: Process completed with exit code 1.
> ```

facing the same thing😕

--- a/packages/better-auth/src/client/client.test.ts
+++ b/packages/better-auth/src/client/client.test.ts
@@ -183,7 +183,9 @@ describe("type", () => {
 				},
 			},
 		});
-		expectTypeOf(client.useComputedAtom).toEqualTypeOf<Readonly<Ref<number>>>();
+		expectTypeOf(client.useComputedAtom).toEqualTypeOf<
+			() => Readonly<Ref<number>>
+		>();
 	});
 	it("should infer resolved hooks svelte", () => {
 		const client = createSvelteClient({

--- a/packages/better-auth/src/client/vue/index.ts
+++ b/packages/better-auth/src/client/vue/index.ts
@@ -33,7 +33,9 @@ type InferResolvedHooks<O extends ClientOptions> = O["plugins"] extends Array<
 							? never
 							: key extends string
 								? `use${Capitalize<key>}`
-								: never]: DeepReadonly<Ref<ReturnType<Atoms[key]["get"]>>>;
+								: never]: () => DeepReadonly<
+							Ref<ReturnType<Atoms[key]["get"]>>
+						>;
 					}
 				: {}
 			: {}

